pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -213,3 +214,53 @@ func TestReload(t *testing.T) {
 	assert.NoError(t, o.Start(ctx, componenttest.NewNopHost()))
 	assert.NoError(t, o.Reload(ctx))
 }
+
+// To be removed when the OpAMP backend correctly advertises its URL.
+// This test is badly written, it reaches into unexported fields to test
+// their contents, but given that it will be removed in a few months, that
+// shouldn't matter too much from a big picture perspective.
+func TestHackSetEndpoint(t *testing.T) {
+	tests := []struct {
+		name         string
+		url          string
+		wantEndpoint string
+	}{
+		{
+			name:         "empty url defaults to config endpoint",
+			wantEndpoint: "https://example.com",
+		},
```

Comment:
I am not sure that this test case indicates valid behaviour. It's a reproduction of the business logic that was encoded in the draft PR, and may need to be changed.